x/ibc/02-client/handler.go

```diff
@@ -104,3 +106,48 @@ func HandlerClientMisbehaviour(k keeper.Keeper) evidencetypes.Handler {
 		return nil
 	}
 }
+
+// HandleClientUpdateProposal is a handler for executing a passed client update proposal
+func HandleClientUpdateProposal(ctx sdk.Context, k keeper.Keeper, p *ibctypes.ClientUpdateProposal) error {
+	clientState, found := k.GetClientState(ctx, p.ClientId)
+	if !found {
+		return types.ErrClientNotFound
+	}
+
+	clientType := clientState.ClientType()
+	switch clientType {
+	case exported.Tendermint:
+		tmClientState := clientState.(*ibctmtypes.ClientState)
+
+		updateClientFlag := false
+		if tmClientState.AllowGovernanceOverrideAfterExpiry && tmClientState.Expired() {
```

Comment:
doesn't something need to be changed here? or the `UpdateClient` function called with a flag? otherwise it seems to me `UpdateClient` will still fail, since the header is still expired

if the testcases aren't catching this, maybe they need to be updated?

Reply:
Agreed, I don't understand how this code would pass in the expired case

Reply:
Here we need to have a manual override function to set the client state without performing the Tendermint verification, but we should still check that the client is not frozen and sanity-check header fields etc.

Reply:
see proposal solution below:
https://github.com/cosmos/cosmos-sdk/pull/7029#issuecomment-676806919